hack/procs/worker.ml

```diff
@@ -24,91 +22,60 @@ external hh_worker_init: unit -> unit = "hh_worker_init"
 (* The maximum amount of workers *)
 let max_workers = 1000
 
-(*****************************************************************************)
-(* The handle is what we get back when we start a job. It's a "future"
- * (sometimes called a "promise"). The scheduler uses the handle to retrieve
- * the result of the job when the task is done (cf multiWorker.ml).
- * Note that the scheduler has to use a handle for that. But the handle
- * is just a trick to get type-checking on workers, a handle is a
- * phantom type, it doesn't really have a value.
- *)
-(*****************************************************************************)
-type 'a handle
+let use_preforked = not Sys.win32
 
 (*****************************************************************************)
-(* Pipes, we need to keep some extra descriptors around to make select work.
- *)
+(* The job executed by the worker. *)
 (*****************************************************************************)
 
-type ('a, 'b) pipe = {
-
-    (* Inputs *)
-    pipe_descr_in  : Unix.file_descr ;
-    pipe_fin       : (unit -> 'a)    ;
-
-    (* Outputs *)
-    pipe_descr_out : Unix.file_descr ;
-    pipe_fout      : ('b -> unit)    ;
-  }
-
-let (make_pipe: unit -> ('a, 'b) pipe) = fun () ->
-  let descr_ic, descr_oc = Unix.pipe() in
-  (* close descriptors on exec so they are not leaked *)
-  Unix.set_close_on_exec descr_ic;
-  Unix.set_close_on_exec descr_oc;
-
-  let ic = Unix.in_channel_of_descr descr_ic in
-  let oc = Unix.out_channel_of_descr descr_oc in
-  let input() = Marshal.from_channel ic in
-  let output data = Marshal.to_channel oc data [Marshal.Closures]; flush oc in
-  { pipe_descr_in = descr_ic;
-    pipe_fin = input;
-    pipe_descr_out = descr_oc;
-    pipe_fout = output;
-  }
+type request =
+  | Request: (('a -> unit) -> unit) -> request
```

Comment:
> > +type request =
> > +  | Request: (('a -> unit) -> unit) -> request

> nit: can we use of instead of :?

Hmm. Let me thinks.

Yes, it mights be replaced by:

```
type serializer = { send : 'a. 'a -> unit }
type request =
  | Request of (serializer -> unit)
```

which is more explicit: the function expects a "serializer" as
argument. This "serializer" should be able to serialize any `'a`.